openstack_controller/datadog_checks/openstack_controller/components/component.py

```diff
@@ -75,10 +76,12 @@ def wrapper(self, *args, **kwargs):
                         tags = argument_value('tags', func, *args, **kwargs)
                         self.check.service_check(self.SERVICE_CHECK, AgentCheck.OK, tags=tags)
                     return result if result is not None else True
-                except HTTPError as e:
-                    self.check.log.error("HTTPError: %s", e.response)
+                except (requests.HTTPError, requests.ConnectionError, requests.Timeout, requests.RequestException) as e:
+                    self.check.log.debug("HTTPError: %s", e.response)
```

Comment:
Always logging `HTTPError` could be misleading , for instance `requests.ConnectionError` could mean a network error:
```
In the event of a network problem (e.g. DNS failure, refused connection, etc), Requests will raise a [ConnectionError](https://requests.readthedocs.io/en/latest/api/#requests.ConnectionError) exception.
```

Reply:
Maybe we could catch all requests exceptions using`requests.exceptions.RequestException` instead? And log the exception type dynamically?

Reply:
Good suggestion!